### PR TITLE
*: simple LOAD DATA escape support

### PR DESCRIPTION
**Pull request by tiancaiamao**
It would fail if we run:

```
cd $GOPATH/src/github.com/go-sql-driver/mysql
MYSQL_TEST_CONCURRENT=1 MYSQL_TEST_ADDR=127.0.0.1:4000 MYSQL_TEST_DBNAME=test go test -v -run=TestLoadData
```

This PR fix that.

@zimulala @shenli

### CODE REVIEW

**Comment by zimulala** [comment]
Add tests.

File: executor/executor_write.go [hunk 1]
```diff
@@ -438,6 +440,40 @@ func (e *LoadDataInfo) InsertData(prevData, curData []byte) ([]byte, error) {
...
+}
+
+func escape(str []byte) []byte {
+	pos := 0
+	for i := 0; i < len(str); i++ {
+		c := str[i]
+		if c == '\\' {
+			if i+1 < len(str) {
+				switch str[i+1] {
+				case 't':
```
Comment (shenli) [inline executor/executor_write.go]: There are many escape chars in MySQL. Should we handle them? Can we use the bytes escape function in scanner?

**Comment by shenli** [comment]
@tiancaiamao CI failed

File: executor/executor_write.go (hunk 1 above)
Comment (tiancaiamao) [inline executor/executor_write.go]: This is very basic support here, while it's much more complex in scanner.
I'm not sure they are same.

File: executor/executor_test.go [hunk 2]
```diff
@@ -628,6 +628,70 @@ func (s *testSuite) TestLoadData(c *C) {
...
+		data1    []byte
+		data2    []byte
+		expected []string
+		restData []byte
+	}
+	// data1 = nil, data2 = nil, fields and lines is default
+	_, err = ld.InsertData(nil, nil)
+	c.Assert(err, IsNil)
+	r := tk.MustQuery(selectSQL)
+	r.Check(nil)
```
Comment (zimulala) [inline executor/executor_test.go]: Line662-666 can be removed.

Comment (zimulala) [inline executor/executor_test.go]: Can you put TestLoadDataEscape into TestLoadData? It makes that line632-line666 can be removed.

File: executor/executor_write.go (hunk 1 above)
Comment (zimulala) [inline executor/executor_write.go]: It is better to add a `TODO:`.

File: executor/executor_test.go (hunk 2 above)
Comment (tiancaiamao) [inline executor/executor_test.go]: The table struct is not the same, so it's hard to reuse all the code or put them in one function.
I'll try to extract some common function.

**Comment by tiancaiamao** [comment]
@zimulala

**Comment by zimulala** [comment]
LGTM

**Comment by tiancaiamao** [comment]
PTAL @shenli

**Comment by shenli** [comment]
LGTM

**Pull request merged**